services/jira/service.go

```
@@ -0,0 +1,239 @@
+}
+
+func (s *Service) Global() bool {
+	return s.global
+}
+
+func (s *Service) RequestPost(method, apiUrl string, postData interface{}) (resp *http.Response, err error) {
+	var post bytes.Buffer
+	enc := json.NewEncoder(&post)
+	err = enc.Encode(postData)
+	if err != nil {
+		return resp, err
+	}
+
+	var jira_url *url.URL
+	jira_url, err = url.Parse(s.url + apiUrl)
+	if err != nil {
+		return resp, err
+	}
+
+	req, err := http.NewRequest(method, jira_url.String(), &post)
+	if err != nil {
+		return resp, err
+	}
+	req.SetBasicAuth(s.username, s.password)
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
+	resp, err = client.Do(req)
+	if err != nil {
+		return resp, err
+	}
+	return resp, err
+}
+
+func (s *Service) Alert(project, issueType, issueFinalStatus, priorityWarn, priorityCrit, entityID, message string, level kapacitor.AlertLevel, details interface{}) error {
+	var priority string
+	var response Response
+	switch level {
+	case kapacitor.WarnAlert:
+		priority = priorityWarn
+		if priority == "" {
+			priority = s.priorityWarn
+		}
+	case kapacitor.CritAlert:
+		priority = priorityCrit
+		if priority == "" {
+			priority = s.priorityCrit
+		}
+	case kapacitor.InfoAlert:
+		return fmt.Errorf("AlertLevel 'info' is currently ignored by the JIRA service")
+	default:
+		priority = "OK"
+	}
+	if project == "" {
+		project = s.project
+	}
+	if issueType == "" {
+		issueType = s.issueType
+	}
+
+	searchData := make(map[string]interface{})
+
+	searchSummary := "\"[" + kapacitor.Product + "] - " + entityID + "\""
+	escapeKeys := [...]string{"[", "]", " "}
+	for _, v := range escapeKeys {
+		searchSummary = strings.Replace(searchSummary, v, "\\\\"+v, -1)
+	}
+	searchData["jql"] = "project = " + project + " AND summary ~ " + searchSummary + " AND status != " + s.issueFinalStatus
+
+	resp, err := s.RequestPost("POST", "/rest/api/2/search", searchData)
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		panic(err.Error())
```

Why panic instead of returning the error?